DigitalLearningSolutions.Web.Tests/Services/EmailVerificationServiceTests.cs

```diff
@@ -101,68 +103,57 @@ public void
-
-            A.CallTo(() => emailService.SendEmail(A<Email>._)).MustHaveHappenedOnceExactly();
-        }
-
-        [Test]
-        public void
-            CreateEmailVerificationHashesAndSendVerificationEmails_updates_hashIds_and_sends_single_email_when_same_centre_email_requires_verification_at_multiple_centres()
-        {
-            // Given
-            var userAccount = UserTestHelper.GetDefaultUserAccount();
-            const int hashId = 1;
-            const string email = "centre@email.com";
-            A.CallTo(() => emailVerificationDataService.CreateEmailVerificationHash(A<string>._, A<DateTime>._))
-                .Returns(hashId);
-
-            // When
-            emailVerificationService.CreateEmailVerificationHashesAndSendVerificationEmails(
-                userAccount,
-                new List<PossibleEmailUpdate>
-                {
                     new PossibleEmailUpdate
                     {
-                        OldEmail = "centre1@email.com",
-                        NewEmail = email,
+                        OldEmail = "old2@email.com",
+                        NewEmail = newEmail2,
                         NewEmailIsVerified = false,
                     },
                     new PossibleEmailUpdate
                     {
-                        OldEmail = "centre2@email.com",
-                        NewEmail = email,
+                        OldEmail = null,
+                        NewEmail = newEmail2,
                         NewEmailIsVerified = false,
                     },
                 },
                 "example.com"
             );
 
             // Then
+            A.CallTo(
+                () => emailVerificationDataService.UpdateEmailVerificationHashIdForPrimaryEmail(
+                    userAccount.Id,
+                    newEmail1,
+                    hashId
+                )
+            ).MustHaveHappenedOnceExactly();
+            A.CallTo(
+                () => emailVerificationDataService.UpdateEmailVerificationHashIdForPrimaryEmail(
+                    userAccount.Id,
+                    newEmail2,
+                    hashId
+                )
+            ).MustHaveHappenedOnceExactly();
+            A.CallTo(
+                () => emailVerificationDataService.UpdateEmailVerificationHashIdForCentreEmails(
+                    userAccount.Id,
+                    newEmail1,
+                    hashId
+                )
+            ).MustHaveHappenedOnceExactly();
             A.CallTo(
                 () => emailVerificationDataService.UpdateEmailVerificationHashIdForCentreEmails(
                     userAccount.Id,
-                    email,
+                    newEmail2,
                     hashId
                 )
             ).MustHaveHappenedOnceExactly();
 
-            A.CallTo(() => emailService.SendEmail(A<Email>._)).MustHaveHappenedOnceExactly();
+            A.CallTo(() => emailService.SendEmail(A<Email>._)).MustHaveHappenedTwiceExactly();

```

Comment:
You should explicitly test that SendEmail is called once for newEmail1 and once for newEmail2